docs/src/developer_documentation/custom_themes.md

```diff
@@ -228,43 +249,45 @@ Then, import your `components.typ` into your theme's entry point:
 
 ### TOC Component
 
-The next three components are a bit more complicated, so we'll be spending a little more time explaining how they work. Each of these components requires some information about the document itself (things like the page numbers of entries, etc.). Normally fetching this data can be rather annoying, but fortunately the Notebookinator has created some utility functions to abstract this.
+The next three components are a bit more complicated, so we'll be spending a little more time explaining how they work. Each of these components requires some information about the document itself (things like the page numbers of entries, etc.). Normally fetching this data can be rather annoying, but fortunately the Notebookinator's constructors fetch this data for you.
 
-To get started with your table of contents, first define a function called `toc` in your `foo/components/toc.typ` file.
+To get started with your table of contents, first define a function called `toc` in your `foo/components/toc.typ` file with the `make-toc` constructor.
 
-However, unlike the `pro-con` function, this function will depend on the `print-toc` utility function. This looks like this:
+Using the `make-toc` constructor we can make a `toc` like this:
 
 ```typ
 // foo/components/toc.typ
 
-// Use this import if you're developing in the Notebookinator directly
-#import "/utils.typ"
-
-// Use this import if you're using the notebookinator as an external dependency
-#import "@local/notebookinator:1.0.1": utils
+// TODO: import utils
 
-#let toc() = utils.print-toc((frontmatter, body, appendix) => {
+#let toc = utils.make-toc((frontmatter, body, appendix) => {
  // ...
 }
 ```
 
-This syntax might look a little weird, so lets break it down. We're defining the function `toc`, which is equal to the `utils.print-toc` function. The `utils.print-toc` function takes 1 argument, and this argument is a function. We're choosing to pass in a [lambda](https://typst.app/docs/reference/foundations/function/#unnamed) function here, since this function only makes sense in the context of the table of contents (we won't need to call it in multiple places).
-
-Inside this lambda we have access to the frontmatter, body, and appendix variables. These variables are all [arrays](https://typst.app/docs/reference/foundations/array/), which dictionaries which all contain the same information as the `ctx` variables from [this](#creating-the-entries) section, with the addition of a `page-number` field, which is an [integer](https://typst.app/docs/reference/foundations/int/).
+Using the constructor gives us access to three variables, `frontmatter`, `body`, and `appendix`. These variables are all [arrays](https://typst.app/docs/reference/foundations/array/), which dictionaries which all contain the same information as the `ctx` variables from [this](#creating-the-entries) section, with the addition of a `page-number` field, which is an [integer](https://typst.app/docs/reference/foundations/int/).
```

Comment:
I think there's a grammar error in the second sentence after the link?

"which dictionaries which all contain..." unless I'm reading it wrong